src/flows/components/Flow.tsx

```diff
@@ -1,15 +1,39 @@
-import React from 'react'
+import React, {useContext} from 'react'
 
 // Components
 import PipeList from 'src/flows/components/PipeList'
 import {PopupDrawer, PopupProvider} from 'src/flows/context/popup'
+import {RunMode, RunModeContext} from 'src/flows/context/runMode'
+import {FlowQueryContext} from 'src/flows/context/flow.query'
 
 // NOTE: requires a FlowProvider and ResultsProvider
-const Flow = () => (
-  <PopupProvider>
-    <PipeList />
-    <PopupDrawer />
-  </PopupProvider>
-)
+const Flow = () => {
+  const {runMode, setRunMode} = useContext(RunModeContext)
+  const {queryAll} = useContext(FlowQueryContext)
+
+  const runPreview = event => {
+    if (event.key === 'Enter' && event.ctrlKey) {
+      if (runMode === RunMode.Preview) {
+        queryAll()
+      } else {
+        setRunMode(RunMode.Preview)
+        queryAll()
+        setRunMode(RunMode.Run)
```

Comment:
not sure about this part

Reply:
I'm open to another way to do it but as far as I could tell queryAll() reads the current state of RunMode so I needed it to be Preview when the queryAll() fires.  I thought about changing RunMode to Preview and leaving it but wasn't sure users would want that behavior.